unstructured_inference/inference/layoutelement.py

```diff
@@ -321,13 +327,44 @@ def get_elements_from_ocr_regions(ocr_regions: List[TextRegion]) -> List[LayoutE
             r.y1,
             r.x2,
             r.y2,
+            None,
             text=r.text,
             type="UncategorizedText",
         )
         for r in merged_regions
     ]
 
 
+def probably_contained(
+    element: Union[LayoutElement, Rectangle],
+    inside: Union[LayoutElement, Rectangle],
+    area_threshold: float = 0.8,
+) -> bool:
+    """This function checks if one element is inside other.
+        If is definetly inside returns True, in other case check
+    if the intersection is big enough."""
+    # Probably can be implemented usigng is_in, however this is based in area overlapping
+    if element.is_in(inside):
+        return True
+
+    intersected_area = element.intersection(inside)
+    if intersected_area:
+        return intersected_area.area >= inside.area * area_threshold
+    return False
+
+
+def separate(region_a: Union[LayoutElement, Rectangle], region_b: Union[LayoutElement, Rectangle]):
+    """Reduce leftmost rectangle to don't overlap with the other"""
+    if not region_a.intersects(region_b):
+        return
+    else:
+        leftmost = region_a if region_a.x1 <= region_b.x1 else region_b
+        other = region_b if region_a.x1 <= region_b.x1 else region_a
+
+        leftmost.x2 = other.x1 - 1
+        leftmost.y2 = other.y2 - 1
```

Comment:
Changing the x should be enough to ensure the rectangles don't overlap, what's happening with the y coordinate change here? Also it doesn't seem right that they should both be `y2`.

Reply:
Looking at @badGarnet 's example, I think we need to carefully evaluate whether this method of deduping makes things better or worse a majority of the time.